frontend/lib/src/components/core/Block/Block.tsx

```diff
@@ -90,7 +93,16 @@ const ChildRenderer = (props: BlockPropsWithoutWidth): ReactElement => {
               node,
             }
 
-            const key = getElementId(node.element) || index.toString()
+            let key = getElementId(node.element)
+            if (key === undefined) {
+              if (node.element.type === "spinner") {
```

Comment:
nit: maybe reuse `isTransient` here to have a single source of truth for what is considered transient.